src/main/java/bookmarked/command/AddCommand.java

```diff
@@ -48,19 +55,108 @@ public void handleCommand() {
 
     /**
      * processAddCommand adds book to the list if not empty
-     * @param newSplitBook the command split by the word 'add' giving the command and description in different arrays
-     * @param listOfBooks the list of books in the library
+     *
      * @throws EmptyArgumentsException throws if there is no description
      */
+    public void processAddCommand(String[] newSplitBook) throws EmptyArgumentsException {
+        // checks if splitQuantity contains only the word "add" or if there are only white spaces after it
+        if (newSplitBook.length < 1) {
+            throw new EmptyArgumentsException();
+        }
 
-    public void processAddCommand(String[] newSplitBook, ArrayList<Book> listOfBooks)
-            throws EmptyArgumentsException {
-        // checks if newSplitBook contains only the word "add" or if there are only white spaces after it
-        if (newSplitBook.length < 1 || newSplitBook[1].isBlank()) {
+        this.splitQuantity = newSplitBook[1].split(" /quantity");
+        if (this.splitQuantity[0].isBlank()) {
             throw new EmptyArgumentsException();
         }
-        Book bookName = new Book(newSplitBook[1].trim());
-        this.listOfBooks.add(bookName);
-        System.out.println("Added " + bookName.getName() + "!");
+
+        try {
+            quantityToAdd = setQuantityToAdd();
+            runAddCommand();
+        } catch (WrongQuantityException e) {
+            Ui.printBlankAddQuantity();
+        } catch (NumberFormatException e) {
+            Ui.printWrongQuantityFormat();
+        } catch (MaxIntNumberException e) {
+            Ui.printMaxNumberMessage();
+        } catch (NegativeQuantityException e) {
+            Ui.printNegativeAddQuantityMessage();
+        }
+    }
+
+
+    public int setQuantityToAdd() throws WrongQuantityException, NumberFormatException,
+            MaxIntNumberException, NegativeQuantityException {
+        //if there is no /quantity argument
```

Comment:
is it a coding standard to have space after "//"? like `// if there is no /quantity argument`